### DISCUSSION TITLE
Is there a way to unlock Windows automatically

### DISCUSSION THREAD

**Discussion by somaraani** [LizardByte/community#254] [Sunshine]
Does anyone know of a way to automatically unlock Windows when running sunshine? I would like to be able to automatically connect even when my PC is sleeping/locked. 

I can see that my startup scripts run, but the Windows Lock screen is on top of it, and I'm not aware of a way to login pragmatically. 

Is there a way similar to how you can configure client credentials when using RDP to automatically unlock the host?

**Comment by ReenigneArcher** [comment]
It's almost impossible to login to Windows with a script/program. I've seen some very old solutions in the past that worked with XP/Vista, but I don't think any of it works on modern versions.

**Comment by Sertan57** [comment]
wouldnt it be possible that susnshine emulates an usb windows helllo device such as a fingerprinter to login? or to login after special controller button combination?

**Comment by jeremie84tr** [comment]
It is possible : [https://www.minitool.com/data-recovery/windows-11-auto-login.html](https://www.minitool.com/data-recovery/windows-11-auto-login.html)

> Way 2. Edit Registry to Enable Windows 11 Auto Login
> Note: Before you edit Windows registry, it’s highly advised you back up Windows Registry first. In case something goes wrong, you can easily restore Windows Registry to a previous setting. Check: [How to backup and restore Windows Registry](https://www.minitool.com/data-recovery/backup-and-restore-registry.html).
> 
> Press Windows + R, type regedit in the Run dialog, and press Enter to open Windows Registry Editor.
> Navigate to the following path: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon.
> Right-click Winlogon and click New -> DWORD (32-bit) Value. Name the new value AutoAdminLogon.
> Double-click AutoAdminLogon and set its value to 1.
> Next, you can right-click Winlogon and click New -> String Value. Name the new value DefaultUserName. Double-click the value and input your user name under Value data. Click OK.
> Then you can right-click Winlogon again and click New -> String Value. Name the new value DefaultPassword. Enter your user account password under Value data and click OK.
> In this way, it will disable the login requirement on your Windows 11 computer and you can log in to your system without password.
> 
> If you want to change the settings back to ask for password again, you can follow the same operation above to delete the created AutoAdminLogon, DefaultUserName, and DefaultPassword keys in Windows Registry Editor.

**Reply by ReenigneArcher** [reply]
That will auto login at system boot. It doesn't allow you to run a script/program to unlock the system at a later point in time.

[a comment by an automated account is not shown]
